src/naive/isoweek.rs

```diff
@@ -160,4 +180,29 @@ mod tests {
         assert_eq!(maxweek.week0(), 0);
         assert_eq!(format!("{:?}", maxweek), MAX_DATE.format("%G-W%V").to_string());
     }
+
+    #[test]
+    fn test_start_day() {
+        let week = NaiveDate::from_ymd(2022, 3, 28).iso_week();
+
+        assert_eq!(week.start_day(Weekday::Mon), NaiveDate::from_ymd(2022, 3, 28));
+        assert_eq!(week.start_day(Weekday::Tue), NaiveDate::from_ymd(2022, 3, 29));
+    }
+
+    #[test]
+    fn test_end_day() {
+        let week = NaiveDate::from_ymd(2022, 3, 28).iso_week();
+
+        assert_eq!(week.end_day(Weekday::Mon), NaiveDate::from_ymd(2022, 4, 3));
+        assert_eq!(week.end_day(Weekday::Tue), NaiveDate::from_ymd(2022, 4, 4));
+    }
+
+    #[test]
+    fn test_bounds() {
+        let date = NaiveDate::from_ymd(2022, 3, 28);
+        let week = date.iso_week();
+
+        assert!(week.bounds(Weekday::Mon).contains(&date));
+        assert!(week.bounds(Weekday::Tue).contains(&date));
```

Comment:
@djc I'm submitting this PR mainly to start a conversation. After adding the `bounds` function I realized that `IsoWeek` might be the wrong place to add this feature. In the context of `test_bounds` the assertion `week.bounds(Weekday).contains(&date)` must be true disregarding which `Weekday` is passed as an argument, returning a week where the original date is out of range might be a little bit counterintuitive. Therefore, I think it would be better to move these 3 functions to the place where `iso_week` is defined since it is required to have access to the `Weekday` of the original date in order to compute the right week boundaries. What do you think?

Reply:
Yeah, that makes sense. Would be nice to add some comments pointing this out so people can find this functionality if they're looking for it in the other place.

Reply:
For any `NaiveDate` the following assertions must always be fulfilled:

```rs
let date = // Any random NaiveDate
for week in Weekday::Mon..Weekday::Sun {
  assert!(date.week_bounds(week).contains(&date));
}
```